### ISSUE TITLE
Go async with database operations

### ISSUE THREAD

**Issue by Mitiko** [Azer0s/Olymp#5]
The general rule of thumb is if you go async, you must go async all the way.

**Comment by Azer0s** [comment]
@Mitiko As soon as everything works semi-stable the database will be read once at startup, loaded into memory and then copied to the p2p database.

**Comment by Mitiko** [comment]
@Azer0s Memory can still get corrupted or be slow, if anything fails, the read might take a little longer and block actions on the main thread.

**Comment by Azer0s** [comment]
@Mitiko true. But in the alpha we will have a separate thread with a ringbuffer queue that writes to disk so we don't block *anything*.
